apps/testnorge-statisk-data-forvalter/src/main/java/no/nav/registre/sdforvalter/consumer/rs/SkdConsumer.java

```diff
@@ -1,68 +1,58 @@
-     */
-    public void createTpsIdenterMessagesInGroup(TpsIdentListe liste, Long playgroup) {
-        Set<SkdRequest> requests = liste.stream().map(SkdRequest::new).collect(Collectors.toSet());
-
-        UriTemplate uriTemplate = new UriTemplate(skdUrl + "/leggTilNyeMeldinger/{playgroup}");
-        RequestEntity<Set<SkdRequest>> requestEntity = new RequestEntity<>(requests, HttpMethod.POST, uriTemplate.expand(playgroup));
-        ResponseEntity<List> responseEntity = restTemplate.exchange(requestEntity, List.class);
-        log.info("ResponseKode: {}", responseEntity.getStatusCode());
-        if (responseEntity.getStatusCode() != HttpStatus.OK) {
-            log.warn("Kunne ikke opprette meldinger på gruppe {} i tpsf", playgroup);
-            throw new HttpServerErrorException(HttpStatus.INTERNAL_SERVER_ERROR);
-        }
+    private final WebClient webClient;
+    private final SkdServiceProperties serverProperties;
+    private final TokenExchange tokenExchange;
+
+    public SkdConsumer(
+            SkdServiceProperties serverProperties,
+            TokenExchange tokenExchange,
+            ExchangeFilterFunction metricsWebClientFilterFunction) {
+
+        this.serverProperties = serverProperties;
+        this.tokenExchange = tokenExchange;
+        this.webClient = WebClient
+                .builder()
+                .baseUrl(serverProperties.getUrl())
+                .filter(metricsWebClientFilterFunction)
+                .build();
     }
 
     /**
      * @param playgroup   AvspillergruppeId som skal spilles av når denne funksjonen er invokert
      * @param environment Miljøet som gruppen skal spilles av til
      */
     public void send(Long playgroup, String environment) {
-        UriTemplate uriTemplate = new UriTemplate(skdUrl + "/startAvspilling/{playgroup}?miljoe={environment}");
-        RequestEntity<String> requestEntity = new RequestEntity<>(HttpMethod.POST, uriTemplate.expand(playgroup, environment));
-        ResponseEntity<SkdResponse> responseEntity = restTemplate.exchange(requestEntity, SkdResponse.class);
-        SkdResponse body = responseEntity.getBody();
-
-        if (responseEntity.getStatusCode() != HttpStatus.OK || body != null && body.getAntallFeilet() != 0) {
-            log.warn("Fikk ikke opprettet alle identene i TPS, burde bli manuelt sjekket for overlapp. " +
-                    "Kan også være mulig at man prøver å initialisere et miljø som er allerede initialisert");
-        }
-        if (body != null) {
-            body.getFailedStatus().forEach(s -> log.error("Status på feilende melding: {}", s));
+        try{
+            var response = tokenExchange.exchange(serverProperties)
+                    .flatMap(accessToken -> new SkdStartAvspillingCommand(
+                            webClient, accessToken.getTokenValue(), playgroup, environment).call())
+                    .block();
+
+            if (isNull(response) || response.getAntallFeilet() != 0) {
```

Comment:
Bruke nonNull() i stedet for != 0

Reply:
Kan nok ikke det da det jeg sjekker er int verdien til getAntallFeilet og ikke om den er null.